Refine content and appearance of the 'Edit View' screen

janfaracik opened the pull request:
This pull request updates the content and minor styling of the 'Edit view' screen.

**What's changed**

* Page title added
* Breadcrumb content aligned with title
* Descriptions added to sections and fields
* Sections reordered
* Recursive jobs are now inset

[![View comparison](https://raw.githubusercontent.com/janfaracik/janfaracik.github.io/add9a80914c0342ea13e4f8c409c953cd9c6f105/comparison/compare.svg)](https://janfaracik.github.io/comparison?before=https://github.com/user-attachments/assets/55399d3a-c533-4bb6-9a90-45784b17fc70&after=https://github.com/user-attachments/assets/55635113-0577-4ec0-82db-368791c975c9&github=https://github.com/jenkinsci/jenkins/pull/9833)

<details>
<summary>
Screenshots
</summary>

<b>Before</b>
<img src="https://github.com/user-attachments/assets/55399d3a-c533-4bb6-9a90-45784b17fc70" />

<b>After</b>
<img src="https://github.com/user-attachments/assets/55635113-0577-4ec0-82db-368791c975c9" />

<b>Before with recursive jobs enabled</b>
<img width="330" alt="image" src="https://github.com/user-attachments/assets/a0f02576-9add-40d5-a396-26829761a55d">

<b>After with recursive jobs enabled</b>
<img width="435" alt="image" src="https://github.com/user-attachments/assets/64eae9d0-4d94-4f0c-b9a2-7c8125a96c52">
</details>

### Testing done

* Works as before

### Proposed changelog entries

- Refine content and appearance of the 'Edit View' screen

### Proposed changelog entries for developers

- `<f:checkbox />` now supports a `description` parameter for setting a description on the checkbox

### Proposed upgrade guidelines

N/A

<!-- Comment:
Leave the proposed upgrade guidelines in the pull request with the "N/A" value if no upgrade guidelines are needed.
The changelog generator relies on the presence of the upgrade guidelines section as part of its data extraction process.
-->

```[tasklist]
### Submitter checklist
- [ ] The Jira issue, if it exists, is well-described.
- [ ] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)). Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [ ] There is automated testing or an explanation as to why this change has no tests.
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [ ] For new APIs and extension points, there is a link to at least one consumer.
```

### Desired reviewers

@mention

<!-- Comment:
If you need an accelerated review process by the community (e.g., for critical bugs), mention @jenkinsci/core-pr-reviewers.
-->

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

janfaracik committed:
Init

janfaracik committed:
Update _style.scss

janfaracik commented:
/label web-ui,rfe

/reviewer @jenkinsci/sig-ux

janfaracik committed:
Merge branch 'master' into improve-view-settings-2

janfaracik committed:
Update configure-entries.jelly

daniel-beck commented on core/src/main/resources/lib/form/checkbox.jelly:
```diff
@@ -87,6 +90,11 @@ THE SOFTWARE.
+    <div class="jenkins-checkbox__description">
+      ${attrs.description}
+    </div>
+  </j:if>
```
Please make sure to document changes like this in the proposed changelog. This is a separate feature for developers.

daniel-beck commented:
Nesting of jobs for the filter looks wrong once it's more than one level, like a tree widget (e.g., Windows Explorer) that only supports a single level. Is this intentional? It's unclear why, to indicate that these entries only exist because of the checkbox?

## Before

<img width="407" alt="Screenshot 2024-10-10 at 14 37 49" src="https://github.com/user-attachments/assets/f3d7b431-c191-4ed1-81fc-a9c6556442bf">


## After

<img width="573" alt="Screenshot 2024-10-10 at 14 37 57" src="https://github.com/user-attachments/assets/23eca419-8c47-4f69-83b7-eae8d0693534">

janfaracik committed:
Merge branch 'master' into improve-view-settings-2

janfaracik committed:
Fix nesting

janfaracik committed:
Update configure-entries-resources.js

janfaracik commented:
> Nesting of jobs for the filter looks wrong once it's more than one level, like a tree widget (e.g., Windows Explorer) that only supports a single level. Is this intentional? It's unclear why, to indicate that these entries only exist because of the checkbox?

Ah good spot, I hadn't considered that. I've updated it so that children of children now nest correctly, I've also updated their labelling so that they don't repeat their parent's name.

**Before**
<img width="480" alt="image" src="https://github.com/user-attachments/assets/efccd8c7-b36b-4972-beab-fa9024142fb7">

**After**
<img width="310" alt="image" src="https://github.com/user-attachments/assets/5d43d1d6-26fe-4a51-b393-0d9eb59803e9">

timja reviewed (approved):
LGTM, I tested with nested folders and all is good:

<img width="276" alt="image" src="https://github.com/user-attachments/assets/095ea4e2-1e19-4cb3-95b8-8889cfbdde0d">

timja committed:
Merge branch 'master' into improve-view-settings-2

timja commented:
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!
